src/passes/I64ToI32Lowering.cpp

```diff
@@ -1526,7 +1520,28 @@ struct I64ToI32Lowering : public WalkerPass<PostWalker<I64ToI32Lowering>> {
   }
 
   void visitSelect(Select* curr) {
-    visitBranching<Select>(curr);
+    if (!hasOutParam(curr->ifTrue)) return;
+    TempVar trueBits = fetchOutParam(curr->ifTrue);
+    TempVar highBits = fetchOutParam(curr->ifFalse);
+    TempVar tmpTrue = getTemp();
+    TempVar tmpFalse = getTemp();
+    Block* result = builder->blockify(
+      builder->makeSetLocal(tmpTrue, curr->ifTrue),
+      builder->makeSetLocal(tmpFalse, curr->ifFalse),
+      builder->makeSelect(
+        curr->condition,
+        builder->blockify(
+          builder->makeSetLocal(
+            highBits,
+            builder->makeGetLocal(trueBits, i32)
+          ),
+          builder->makeGetLocal(tmpTrue, i32)
+        ),
+        builder->makeGetLocal(tmpFalse, i32)
+      )
+    );
+    replaceCurrent(result);
+    setOutParam(result, std::move(highBits));
```

Comment:
I may be missing something, but shouldn't the out param be a select over either the high bits of the ifTrue or the ifFalse?

Reply:
Sorry, highBits (now falseBits) will contain that. I got carried away with optimizations and pushed wrong commit. Changed makeIf to makeSelect, reverted this.